tests/baselines/reference/classUsedBeforeInitializedVariables.js

```diff
@@ -0,0 +1,102 @@
+//// [classUsedBeforeInitializedVariables.ts]
+class Test {
+    p1 = 0;
+    p2 = this.p1;
+    p3 = this.p4;
+    p4 = 0;
+
+    directlyAssigned: any = this.directlyAssigned;
+
+    withinArrowFunction: any = () => this.withinArrowFunction;
+
+    withinFunction: any = function () {
+        return this.withinFunction;
+    };
+
+    withinObjectLiteral: any = {
+        [this.withinObjectLiteral]: true,
+    };
+
+    withinObjectLiteralGetterName: any = {
+        get [this.withinObjectLiteralGetterName]() {
+            return true;
+        }
+    };
+
+    withinObjectLiteralSetterName: any = {
+        set [this.withinObjectLiteralSetterName](_: any) {}
+    };
+
+    withinClassDeclarationExtension: any = (class extends this.withinClassDeclarationExtension { });
+
+    // These error cases are ignored (not checked by control flow analysis)
+
+    assignedByArrowFunction: any = (() => this.assignedByFunction)();
+
+    assignedByFunction: any = (function () {
+        return this.assignedByFunction;
+    })();
+}
+
+
+//// [classUsedBeforeInitializedVariables.js]
+var __extends = (this && this.__extends) || (function () {
+    var extendStatics = function (d, b) {
+        extendStatics = Object.setPrototypeOf ||
+            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
+            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
+        return extendStatics(d, b);
+    };
+    return function (d, b) {
+        extendStatics(d, b);
+        function __() { this.constructor = d; }
+        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
+    };
+})();
+var Test = /** @class */ (function () {
+    function Test() {
+        var _a, _b, _c;

```

Comment:
this line moved after I merged from master and re-ran the tests, but I think it's fine.